sdk/lib/rtl/amd64/unwind.c

```diff
@@ -687,3 +687,72 @@ RtlRaiseException(IN PEXCEPTION_RECORD ExceptionRecord)
     RtlRaiseStatus(Status);
 }
 
+static
+VOID
+RtlpCaptureNonVolatileContextPointers(
+    _Out_ PKNONVOLATILE_CONTEXT_POINTERS NonvolatileContextPointers,
+    _In_ ULONG64 TargetFrame)
+{
+    CONTEXT Context;
+    PRUNTIME_FUNCTION FunctionEntry;
+    ULONG64 ImageBase;
+    PVOID HandlerData;
+    ULONG64 EstablisherFrame;
+    KPROCESSOR_MODE Mode;
+
+    /* Zero out the nonvolatile context pointers */
+    RtlZeroMemory(NonvolatileContextPointers, sizeof(*NonvolatileContextPointers));
+
+    /* Capture the current context */
+    RtlCaptureContext(&Context);
+
+    /* Use the highest bit in RIP as mode flag */
+    Mode = Context.Rip >> 63;
```

Comment:
RtlpGetMode() ? This looks fishy.

Reply:
I simplified it and added some ASSERTs instead.